Ecierge.Uno.Controls/NullableNumberBox.cs

```diff
@@ -0,0 +1,525 @@
+namespace Ecierge.Uno.Controls;
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.Globalization.NumberFormatting;
+using static CommunityToolkit.WinUI.TextBoxExtensions;
+
+[TemplatePart(Name = PartHeaderContentPresenter, Type = typeof(FrameworkElement))]
+[TemplatePart(Name = PartCheckBox, Type = typeof(CheckBox))]
+[TemplatePart(Name = PartNumberBox, Type = typeof(NumberBox))]
+public sealed partial class NullableNumberBox : Control
+{
+    private const string PartHeaderContentPresenter = "HeaderContentPresenter";
+    private const string PartNumberBox = "NumberBox";
+    private const string PartCheckBox = "CheckBox";
+
+    private bool isChecked;
+    private NumberBox? numberBox;
+    private CheckBox? checkBox;
+
+    public event TypedEventHandler<NullableNumberBox, NumberBoxValueChangedEventArgs>? ValueChanged;
+
+    #region AcceptsExpression
+
+    /// <summary>
+    /// Identifies the AcceptsExpression dependency property.
+    /// </summary>
+    public static readonly DependencyProperty AcceptsExpressionProperty =
+        DependencyProperty.Register(nameof(AcceptsExpression), typeof(bool), typeof(NullableNumberBox),
+            new(NumberBox.AcceptsExpressionProperty.GetMetadata(typeof(NumberBox)).DefaultValue));
+
+    /// <summary>
+    /// Toggles whether the control will accept and evaluate a basic formulaic expression
+    /// entered as input.
+    /// </summary>
+    public bool? AcceptsExpression
+    {
+        get => (bool?)GetValue(AcceptsExpressionProperty);
+        set => SetValue(AcceptsExpressionProperty, value);
+    }
+
+    #endregion AcceptsExpression
+
+    #region Header
+
+    /// <summary>
+    /// Header Dependency Property
+    /// </summary>
+    public static readonly DependencyProperty HeaderProperty =
+        DependencyProperty.Register(nameof(Header), typeof(object), typeof(NullableNumberBox),
+             new PropertyMetadata(NumberBox.HeaderProperty.GetMetadata(typeof(NumberBox)).DefaultValue));
+           // new PropertyMetadata((object?)null, new PropertyChangedCallback(OnHeaderChanged)));
+
+    /// <summary>
+    /// Gets or sets the Header property. This dependency property
+    /// indicates header.
+    /// </summary>
+    public object? Header
+    {
+        get => (object?)GetValue(HeaderProperty);
+        set => SetValue(HeaderProperty, value);
+    }
+
+    #endregion
```

Comment:
В закрывающих тоже название должно быть
```suggestion
    #endregion Header
```